packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -227,6 +238,48 @@ const App = (props) => {
         })
     }, [])
 
+    const {usid} = useUsid()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const deleteShopperContext = useShopperContextsMutation('deleteShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {
+            parameters: {usid: usid, siteId: site.id}
+        },
+        {
+            enabled: !isServer,
+            onError: async () => {
+                await createShopperContext.mutateAsync({
```

Comment:
Please leave a comment why we are creating shopper context onError. Is there a way for us to distinguish between shopper context not existed and other errors?